doc/database/hierarchy.md

```diff
@@ -0,0 +1,250 @@
+  Family <|-- Genus
+```
+
+Each of these entities can have their own attributes.
+E.g., a variety under a specie can have different height than the specie that it belongs to.
+
+## Unique Name
+
+In the database we have a column `unique_name`.
+The unique name are, either for abstract plants:
+
+- single latin word to specify "family", which always ends on: `-aceae`
+- single latin word to specify "genus"
+
+Or is built up by several words (for concrete plants):
+
+- single latin word to specify "genus"
+- single latin word to specify "specie"
+- optional single latin word to specify "variety"
+- optional several words in single quotes to specify "cultivar", which starts with a capital letter
+
+E.g. `Brassica oleracea italica 'Ramoso calabrese'`
+
+### Hybrid names
+
+Hybrid names are built up differently.
+Either two parent binomials, separated by a "x" or "×" or a given binomial, with or without an intercalated "×"
+(see [Wikipedia](https://en.wikipedia.org/wiki/Hybrid_name)).
+So the name does not necessarily say if a plant is a hybrid.
+
+### Rendering
+
+The unique name must be rendered:
+
+- Latin name in _italics_.
+- Cultivar name normal font in single quotes (as in database).
+
+E.g.: _Brassica oleracea italica_ 'Ramoso calabrese'
+
+### Rules
+
+We know about names (abstract and concrete, including hybrid):
+
+- If it contains more than one word, it is a concrete plant.
+- If it contains only one word, it is an abstract plant (family or genus).
+- All entries with 2 words are a specie (de: Art),
+- All entries with 3 or more words are either:
+  - variety (de: Varietät) if all is spelled italic, or
+  - are cultivar (de: Sorte) if last part of the name (can be more than 1 word!) is not latin, not italic, is in 'single quotes'
+- All entries with a single x between the words are hybrid.
+  We treat them like specie.
+
+## Attributes
+
+We prefer strongly-typed data, e.g.:
+
+- enums
+- array of enums
+- numbers
+
+Columns that contain text should be postfixed with:
+
+- `_en` for English text
+- `_de` for German text
+
+Plants are additionally classified as:
+
+- is_concrete_plant (is a concrete plant as opposite to an abstract plant)
+- is_tree (as search help within the tree layer)
+
+## Mappings
+
+### from Permapeople
+
+Column names should be:
+
+- `unique_name` (see above), instead of `binomial_name`
+- `common_name_en` (still with arrays of Text as content)
+- `common_name_de` should be from Wikipedia (still with arrays of Text as content), `german_name` gets removed
+- `hardiness_zone` (usda in documentation)
+- `light_requirement` (and not `sun`)
+- `water_requirement` (and not `water`)
+- `soil_texture` (and not `soil_type`)
+- `height` (and not `mature_size_height`, mature height in documentation)
+- `width` (and not `mature_size_width`, mature width in documentation)
+- `propagation_method` (and not `propagation`)
+- `growth_rate` (and not `growth`)
+- `has_drought_tolerance`
+- `id` should be `permapeople_id`
```

Comment:
Thank you, a similar suggestion also came from @aidnurs. A small problem might be that some rows are mixed, especially permapeople and practicalplants have big overlap (about 6000 plants). But as practicalplants is dead, we can simply let `external_id` and `external_url` ignore practicalplants. The overlap between permapeople and reinsaat hopefully isn't big, as reinsaat nearly exclusively has cultivars. Some of the data from reinsaat about species, however, should end up in the rows of permapeople, so that future new cultivars get more complete data.

I changed now to `external_` as proposed by @Benjaminimal and @aidnurs  in be040a822a7c6c157c3e984ddebb99ed47545d86 but if the above mentioned problem hits us too hard, we can keep `reinsaat_` and `permapeople_` specific columns for id/urls (then source is not needed as the presence of id/urls already says the source).

@aidnurs please keep us up to date about how many rows are mixed in which ways.